module/VuFind/src/VuFind/Cover/LoaderFactory.php

```diff
@@ -68,8 +68,10 @@ public function __invoke(
         if (!empty($options)) {
             throw new \Exception('Unexpected options sent to factory.');
         }
-        $cacheDir = $container->get(\VuFind\Cache\Manager::class)
-            ->getCache('cover')->getOptions()->getCacheDir();
+        $cacheOptions = $container->get(\VuFind\Cache\Manager::class)
+            ->getCache('cover')->getOptions();
+        $cacheDir = method_exists($cacheOptions, 'getCacheDir')
+            ? $cacheOptions->getCacheDir() : null;
```

Comment:
`getCacheDir` is specific to FileSystemOptions, used by the filesystem adapter, and not present in the parent class AdapterOptions, which is what the BlackHole cache uses.

Given the supposed intent of BlackHole to be an invisible substitute, I don't like having to effectively check here for the presence of that cache directory.  I played around for a while with subclassing AdapterOptions to provide a dummy implementation of getCacheDir to return null; which after all works fine.  I did that and changed Cache\Manager->createNoCache() to use my subclass, but I couldn't get Laminas to find a factory for it, since it's using its own configuration not the mapping in the VuFind module's module.config.php.  So anyway...trying this approach.

Reply:
I think the simplicity of this approach seems reasonable. There should be a way to override the Laminas factories via module.config.php -- probably a matter of just figuring out exactly which key(s) to use -- but that seems overcomplicated when we can do things this way.

However, I think it may be better to use `is_callable([$cacheOptions, 'getCacheDir'])` instead of `method_exists($cacheOptions, 'getCacheDir')` to make sure that callability is accounted for instead of method existence. This is especially important in case any of the option classes rely on magic methods like `__call`, which is a fairly common pattern when wrapping options in objects.

Reply:
Done